trie/branches.py

```diff
@@ -0,0 +1,190 @@
+from trie.trie import (
+    BinaryTrie,
+)
+from trie.exceptions import (
+    InvalidKeyError,
+)
+from trie.constants import (
+    BLANK_HASH,
+    KV_TYPE,
+    BRANCH_TYPE,
+    LEAF_TYPE,
+    BYTE_0,
+)
+from trie.validation import (
+    validate_is_bytes,
+    validate_is_bin_node,
+)
+
+from trie.utils.sha3 import (
+    keccak,
+)
+from trie.utils.binaries import (
+    encode_to_bin,
+)
+from trie.utils.nodes import (
+    parse_node,
+)
+
+
+def if_branch_exist(db, root_hash, key_prefix):
+    """
+    Given a key prefix, return whether this prefix is
+    the prefix of an existing key in the trie.
+    """
+    validate_is_bytes(key_prefix)
+
+    return _if_branch_exist(db, root_hash, encode_to_bin(key_prefix))
+
+
+def _if_branch_exist(db, node_hash, key_prefix):
```

Comment:
*naming nitpick*

`check_if_branch_exists` might convey the functionality of this function a bit better.